pupil_src/shared_modules/glfw.py

```diff
@@ -761,3 +790,46 @@ def get_framebuffer_scale(window) -> float:
 def window_coordinate_to_framebuffer_coordinate(window, x, y, cached_scale=None):
     scale = cached_scale or get_framebuffer_scale(window)
     return x * scale, y * scale
+
+
+def get_window_content_rect(window) -> _Rectangle:
+    x, y = glfwGetWindowPos(window)
+    w, h = glfwGetWindowSize(window)
+    return _Rectangle(x=x, y=y, width=w, height=h)
+
+
+def get_window_frame_rect(window) -> _Rectangle:
+    content_rect = get_window_content_rect(window)
+    frame_edges = glfwGetWindowFrameSize(window)
+    return _Rectangle(
+        x=content_rect.x - frame_edges.left,
+        y=content_rect.y - frame_edges.top,
+        width=content_rect.width + frame_edges.left + frame_edges.right,
+        height=content_rect.height + frame_edges.top + frame_edges.bottom,
+    )
+
+
+def get_window_title_bar_rect(window) -> _Rectangle:
+    frame_rect = get_window_frame_rect(window)
+    frame_edges = glfwGetWindowFrameSize(window)
+    return _Rectangle(
+        x=frame_rect.x, y=frame_rect.y, width=frame_rect.width, height=frame_edges.top
```

Comment:
I don't understand why `height=frame_edges.top`. If I am not mistaken `top` corresponds to
> in screen coordinates, [...] the top edge of the window frame

Reply:
`top` here represents the distance between the top edge of the window frame and the top edge of the window content, which in case of `top` should be equal or slightly bigger than then height of the window title bar.

Reply:
Ah, makes sense. Thanks for clarifying!